### ISSUE TITLE
Create session link so auto-saved session can be opened from bookmark toolbar

### ISSUE THREAD

**Issue by kaanchan** [sienori/Tab-Session-Manager#700]
**Is your feature request related to a problem? Please describe.**
I keep different windows open for different projects, each of which has multiple tabs (but never multiple windows per project). I don't know of a simple way to add a bookmark/link for a project/window session so I can open and close as I work and stop working on a project.

**Describe the solution you'd like**
I'd like to be able to assign a specific window a specific session (which will be auto-saved, taggable, ...).
I'd like to get a link back which I can save/bookmark, and be able to place in a bookmark folder.
I'd like to be able to click/select that session link and have that window open and session restored. 
I'd like the session to be auto-saved when window closes.
It would be awesome if I could add nameable save-points to a manual save!

**Describe alternatives you've considered**
The only alternative I can think of is to have all my session/project windows at the same time.

**Additional context**
The above would really turn this add-on into a ultimate project management tool for Firefox research and browsing.

**Comment by perXautomatik** [comment]
[ToManyTabs](https://chrome.google.com/webstore/detail/toomanytabs-for-chrome/amigcgbheognjmfkaieeeadojiibgbdp?hl=en) used to store all your tabs into a subfolder of your bookmarks, this was sometimes rather messy but i do see the apeal nowdays if the omnibox in for example chrome could pick up stored sessions and suggest it as you type.

one of the problems is same as mentioned, it might very quickly get cluttered. 

don't know how dificult it would be to make stored sessions a "Live bookmark" but it would be absolutely amazing for the purpus of indexing

**Comment by guraltsev** [comment]
This would be fantastic also for manual sessions. I have many manual sessions saved and I would like to be able to start the browser with that window programmatically. I would like to have a link I could double click on my desktop that would open the browser with the "project1" session.

**Comment by kaanchan** [comment]
> one of the problems is same as mentioned, it might very quickly get cluttered.

How do you imagine or see it would get cluttered? Due to too many tabs? I use a tree tab add-on which nests. Those could be converted to nested folders or some equivalent hierarchical system, maybe? Also, I imagine that as things get cluttered, I have to clean up from time to time :)

**Comment by perXautomatik** [comment]
> How do you imagine or see it would get cluttered? Due to too many tabs? 

Yeah, i currently save every closed window as a session, my way to collect my thoughts, and adding 30-90 tabs a day every day, the folders would fill up quickly, 

what would that be to a relational database, probably not too much.

Though, 
When I have been moving bookmarks between browsers, summing up to several thousands, I've noticed some performance issues,
i suppose neither google nor Mozilla had that in mind when they optimized the bookmark experience for the browsers.


>I use a tree tab add-on which nests. Those could be converted to nested folders or some >equivalent hierarchical system, maybe? Also, I imagine that as things get cluttered, I have to >clean up from time to time :)

Ooh! Might I ask which three you use? 
I'm painfully aware my current solutions for session managing is lacking, so always interested in new extensions I might add to my tool chain,

>+1 for indexing and searching.
I was thinking of a possible MongoDB server storing Tab-session backups, so I could parse them, but thinking more and more about what bookmarks can do.
